### ISSUE TITLE
Move fswatcher for gatsby-source-filesystem to worker

### ISSUE THREAD

**Issue by KyleAMathews** [gatsbyjs/gatsby#5499]
Looks like it improves performance quite a bit https://github.com/parcel-bundler/parcel/pull/1368

**Comment by KyleAMathews** [comment]
Cool! Thanks!

**Comment by KyleAMathews** [comment]
Watching lots of files still seems to be causing trouble and I'm keen to see if this technique improves build speeds.

Also Parcel has published their wrapper around Chokidar as a separate plugin so would be straightforward to try @parcel/watcher

**Comment by KyleAMathews** [comment]
https://www.npmjs.com/package/@parcel/watcher

**Comment by KyleAMathews** [comment]
To test this, we could create a site with e.g. 1000 and 10000 markdown files and see how sourcing files compare with this watcher vs. our existing watcher in gatsby-source-filesystem.

**Comment by LekoArts** [comment]
So I tried to use `@parcel/watcher` -- without success sadly. Firstly, the documentation is currently really sparse and I used a WIP PR from Parcel and their tests to figure out some things. 
Secondly my tests with a Gatsby site show that the Watcher only registers changed files which resulted in no nodes created at all on the startup of `gatsby develop`. Their snapshot feature also doesn't work for that usecase (paired with @pieh on this one). So we'd need to implement another solution to grab the initial files and then only use `@parcel/watcher` for changed files.
Lastly it is still in alpha stage which @pieh and I realized when we had the classic "it works on my machine" situation -- the watcher worked for him during `gatsby develop`, for me not at all (he's on Mac, I'm on Win10 Linux Subsystem).

So I think we should press pause on this feature request and revisit it later.

My current work so far can be found [here](https://github.com/LekoArts/gatsby/blob/e78b03512cd0c1dfc11dd08234b339af88bfdbeb/packages/gatsby-source-filesystem/src/gatsby-node.js).
